cmd/upgrade.go

```diff
@@ -0,0 +1,133 @@
+package cmd
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/spf13/cobra"
+)
+
+var upgradeCmd = &cobra.Command{
+	Use:   "upgrade",
+	Short: "Upgrade osdctl",
+	Long:  "Fetch latest osdctl from GitHub and replace the running binary",
+	RunE:  upgrade,
+}
+
+func upgrade(cmd *cobra.Command, args []string) error {
+	latest, err := getLatestVersion()
+	if err != nil {
+		return err
+	}
+	latestWithoutPrefix := strings.TrimPrefix(latest, "v")
+	// check if an upgrade is necessary
+	currentSemVer := semver.New(Version)
+	latestSemVer := semver.New(latestWithoutPrefix)
+	if currentSemVer.LessThan(*latestSemVer) {
+		// upgrade necessary
+		client := http.Client{
+			Timeout: time.Second * 60,
+		}
+
+		addr := fmt.Sprintf(versionAddressTemplate, latestWithoutPrefix, latestWithoutPrefix,
+			parseGOOS(runtime.GOOS), parseGOARCH(runtime.GOARCH))
+
+		req, err := http.NewRequest(http.MethodGet, addr, nil)
+		if err != nil {
+			return err
+		}
+
+		res, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+
+		gzf, err := gzip.NewReader(res.Body)
+		if err != nil {
+			return err
+		}
+
+		tr := tar.NewReader(gzf)
+		for {
+			f, err := tr.Next()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return err
+			}
+			if f.Name == "osdctl" {
```

Comment:
```suggestion
			if f.Name != "osdctl" {
                               continue
                        }
                        ....
```
removes one level of nesting